android: Implement about game dialog

kleidis opened the pull request:
This PR adds a new "about game dialog" menu when you long press on a game entry within the home screen instead of just going to the cheats menu

I don't know how useful you find this now but it can be expanded later on the future with more buttons and possibly a GameID string.

Menu screenshot:

<img src="https://github.com/user-attachments/assets/4b5ecc87-0d8b-4fa3-a517-898238131600" height=500px/>

Ishan09811 committed:
android: Implement about game dialog

OpenSauce04 commented on src/android/app/src/main/res/layout/game_about_dialog.xml:
```diff
@@ -0,0 +1,131 @@
+                android:layout_marginStart="8dp"
+                android:textColor="?attr/colorAccent"
+                app:icon="@drawable/ic_settings"
+                app:iconGravity="textStart" />
```
The placement of this button seems to indicate that it should open game-specific settings, however it instead opens the regular shared settings menu. What's the intention here?

OpenSauce04 commented on src/android/app/src/main/res/layout/game_about_dialog.xml:
```diff
@@ -0,0 +1,131 @@
+            android:layout_marginTop="8dp"
+            android:text="@string/add_ons"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toBottomOf="@+id/horizontal_layout" />
```
Why is the phrase "Add-ons" used here? It makes little sense in this context.

OpenSauce04 commented on src/android/app/src/main/res/layout/game_about_dialog.xml:
```diff
@@ -0,0 +1,131 @@
+                app:layout_constraintStart_toStartOf="@id/game_title"
+                app:layout_constraintTop_toBottomOf="@id/game_title"
+                tools:text="Game Author" />
+
```
In order to pad out this space with some more useful information, I would recommend also displaying the game's ID and region.

I also think adding the game's compatibility rating here would be nice too, but I would consider that out of the scope of this PR as further changes will need to be made for this to be possible.

OpenSauce04 commented on src/android/app/src/main/res/values/strings.xml:
```diff
@@ -432,9 +432,17 @@
+    <string name="add_ons">Add-ons</string>
+    <string name="delete_save">Delete save</string>
+    <string name="import_save">Import</string>
+    <string name="export_save">Export</string>
```
There appear to be strings here for functionality that doesn't exist in the about dialog. Are these planned, or have they been cut from the feature?

OpenSauce04 committed:
Nitpick code corrections

OpenSauce04 committed:
Renamed `game_about_dialog` to `dialog_about_game` to follow naming convention

OpenSauce04 committed:
Removed unused strings

OpenSauce04 committed:
Updated about game dialog layout

OpenSauce04 committed:
Fixed about game dialog displaying incorrectly in landscape orientation

Commit:
Re-arrange UI Elements

- Removed file name from card_game and replaced it with publisher and region
- Removed publisher from about game dialog and added GameID and file name
- Renamed xml id's for the about dialog xml to make more sense

OpenSauce04 committed:
Minor about game dialog tweaks

- Added game company and game region strings to about game dialog
- Swapped game ID and game file strings in about game dialog

OpenSauce04 committed:
Updated license header

OpenSauce04 committed:
Add prefixes to ID and filename strings in about game dialog
